File: src/main/java/org/chillout1778/subsystems/Controls.java
package org.chillout1778.subsystems;

import edu.wpi.first.wpilibj.Joystick;
import edu.wpi.first.wpilibj.XboxController;
import org.chillout1778.Robot;
import org.chillout1778.Constants;

public class Controls {
    private static Controls instance;
    public static Controls getInstance() {
        if (instance == null) instance = new Controls();
        return instance;
    }

    private final Joystick driver = new Joystick(0);

    private static double deadzone(double n) {
        if (Math.abs(n) < 0.1) {
            return 0.0;
        } else {
            return n;
        }
    }

    public double driveFactor() {
        return -0.5 * deadzone(driver.getRawAxis(Constants.Controls.driveAxisID));
    }

    public double turnFactor() {
        return 0.5 * deadzone(driver.getRawAxis(Constants.Controls.turnAxisID));
    }
}

File: src/main/java/org/chillout1778/subsystems/Controls.kt
package org.chillout1778.subsystems

import edu.wpi.first.wpilibj.Joystick

object Controls {
    val driver = Joystick(0)
    fun x() = driver.getRawAxis(2)
    fun y() = driver.getRawAxis(3)
    fun rot() = driver.getRawAxis(0)
    // Autoalign: driver.getRawButton(1)
}
